serde/src/private/de.rs

```diff
@@ -2287,11 +2308,17 @@ mod content {
             )
         }
 
-        fn visit_seq<S>(self, _: S) -> Result<(), S::Error>
+        fn visit_seq<S>(self, mut seq: S) -> Result<(), S::Error>
         where
             S: SeqAccess<'de>,
         {
-            Ok(())
+            match tri!(seq.next_element()) {
```

Comment:
This behaves quite differently from `IgnoredAny.visit_map`. I think the behaviour should be consistent, as in, iterate over the entire sequence and ignore its values.

Reply:
I tried that initially, but that failed other tests and in general not what you want. The unit / unit struct represented in sequence as nothing, so we need to ensure that sequence is empty. This is consistent with normal behavior where struct deserialization from a sequence expects exact number of values, and those fact that flattened unit / unit struct considered as equal to the struct without fields.

Reply:
Actually, the `unit_variant_with_unknown_fields` is a test that failed if consume the whole sequence here.
https://github.com/serde-rs/serde/blob/3aca38d2d3c9b5dba9bcb25be2e16bde6678dbf6/test_suite/tests/test_enum_internally_tagged.rs#L1447-L1456

Reply:
> The unit / unit struct represented in sequence as nothing

Hm but "nothing" should be pretty different conceptually from "ignored any". I'd expect a custom check just for the nothing case, whereas ignored any should be able to consume anything thrown at it silently.

Reply:
This code tries to read something, doesn't matter what. We expect an empty sequence, so if it contains some element, we fail.

Reply:
Nevermind, I'm sleepy - I thought you're changing how IgnoredAny works everywhere. I've expanded the context of the diff and I see this is a change on this one specific visitor.

Please disregard my original comment 🤦‍♂️

Although I now wonder if `visit_map` should be changed to check length as well.

Reply:
By default maps in serde allows unknown keys and when unit is flattened, all keys become unknown. But you're right -- in case of `#[serde(deny_unknown_fields)]` we should return error if map not empty. That's idea for another PR!